pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -453,7 +458,9 @@ synchronized public void close() {
         }
 
         if (null != javaInstance) {
+            Thread.currentThread().setContextClassLoader(functionClassLoader);
             javaInstance.close();
+            Thread.currentThread().setContextClassLoader(instanceClassLoader);
```

Comment:
need `try ... finally ...`  to make sure the classloader is reset correctly

Reply:
done